code/modules/mob/living/simple_animal/hostile/necro.dm

```diff
@@ -96,11 +130,313 @@
 		ticker.mode.update_necro_icons_added(Controller)
 		ticker.mode.update_all_necro_icons()
 		ticker.mode.risen.Add(Controller)
-	name += " ([rand(1,1000)])"
 
-/mob/living/simple_animal/hostile/necro/copy/ListTargets()
-	. = ..()
-	return . - creator
+	if(name == initial(name) && !unique_name)
+		name += " ([rand(1,1000)])"
+
+/mob/living/simple_animal/hostile/necro/zombie/Life()
+	..()
+	/*TODO
+	First, check if the zombie can potentially evolve
+	Have the zombie move to a corpse and start chewing at it
+	If neither of these things are applicable, break some lights to set the mood
+	Otherwise, start wandering and bust down some doors to find more food
+	*/
+	if(!isUnconscious())
+		if(stance == HOSTILE_STANCE_IDLE) //Not doing anything at the time
+			var/list/can_see = view(src, vision_range)
+			if(!busy)
+				if(can_evolve)//Can we evolve, and have we fed
+					busy = EVOLVING
+					check_evolve()
+					busy = 0
+				if((health < maxHealth) || (maxHealth < health_cap)) //Is there something to eat in range?
+					for(var/mob/living/carbon/human/C in can_see) //If so, chow down
+						if(C.isDead() && !busy && check_edibility(C))
+							Goto(C, speed)
+							busy = MOVING_TO_TARGET
+							give_up(C) //If we're not there in 10 seconds, give up
+
+							if(C.Adjacent(src) && busy != EATING) //Once we've finally caught up
+								busy = EATING
+								eat(C)
+				if(!busy && break_doors != CANT)//So we don't try to eat and open doors
+					/*for(var/obj/machinery/light/L in can_see)
+						if(L.status != 1 && L.status !=2)
+							Goto(L, speed)
+							busy = MOVING_TO_TARGET
+							if(L.Adjacent(src) && busy != SMASHING_LIGHT)
+								L.attack_animal(src)*/
+
+					for(var/obj/machinery/door/D in can_see)//Else, let's open some doors and see what's around
+						if(can_open_door(D) && !busy)
+							Goto(D, speed)
+							busy = MOVING_TO_TARGET
+							give_up(D)
+
+							if(D.Adjacent(src) && busy != OPENING_DOOR)
+								busy = OPENING_DOOR
+								force_door(D)
+
+		else
+			busy = 0
+			stop_automated_movement = 0
```

Comment:
If it falls unconscious this wont stop its movement, you also have to cancel the automated walk_to that Goto() uses